timelock-api/src/main/conjure/timelock-api.yml

```diff
@@ -217,6 +217,40 @@ types:
       LeaderTimes:
         fields:
           leaderTimes: map<Namespace, LeaderTime>
+      TimestampName:
+        alias: string
+        safety: safe
+      NamedMinTimestampLeaseRequest:
+         fields:
+           timestampName: TimestampName
+           requestId:
+             type: uuid
+             safety: safe
+           numFreshTimestamps:
+             type: integer
+             safety: safe
+      NamedMinTimestampLeaseRequests:
+         alias: list<NamedMinTimestampLeaseRequest>
+      MultiClientNamedMinTimestampLeaseRequest:
+         alias: map<Namespace, NamedMinTimestampLeaseRequests>
+      NamedMinTimestampLeaseResponse:
+         fields:
+           minLeased: Long
+           lock: LockToken
+           lease: Lease
+           freshTimestamps: ConjureTimestampRange
+      NamedMinTimestampLeaseResponses:
+         alias: list<NamedMinTimestampLeaseResponse>
+      MultiClientNamedMinTimestampLeaseResponse:
+         alias: map<Namespace, NamedMinTimestampLeaseResponses>
+      GetMinLeasedNamedTimestampRequests:
+         alias: set<TimestampName>
```

Comment:
Generally always cheaper to stick a list?

Reply:
Fixing